src/io.rs

```diff
@@ -341,6 +361,38 @@ impl Write for &'_ mut [u8] {
     }
 }
 
+/// Converts a [`std::io::Write`] into a [`manticore::io::Write`].
+///
+/// [`write_bytes()`] is implemented by simply calling [`write()`] repeatedly
+/// until every byte is written; [`manticore::io::Write`] should be implemented
+/// directly if possible.
+///
+/// This type is provided instead of implementing [`manticore::io::Write`]
+/// directly for every [`std::io::Write`] due to trait coherence issues
+/// involving the blanket impl on `&mut _`.
+///
+/// [`std::io::Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
+/// [`manticore::io::Write`]: trait.Write.html
+/// [`write_bytes()`]: trait.Write.html#tymethod.write_bytes
+/// [`write()`]: https://doc.rust-lang.org/std/io/trait.Write.html#tymethod.write
+#[cfg(feature = "std")]
+pub struct StdWrite<W>(pub W);
```

Comment:
I think this need to be `pub &'a mut W`, as otherwise we'd move `W` into `StdWrite`, which means that we couldn't use it anymore. Or am I missing something?

Reply:
I ended up with the following implementation for testing:
```
pub struct StdWrite<'a, W>(pub &'a mut W);

impl<'a, W: std::io::Write> Write for StdWrite<'a, W> {
    fn write_bytes(&mut self, buf: &[u8]) -> Result<(), Error> {
      let mut idx = 0;

      while idx < buf.len() {
        match self.0.write(&buf[idx..]) {
          Err(_) => return Err(Error::Internal),
          Ok(len) => idx = idx + len,
        }
      }

      Ok(())
    }
}
```

This can now be used like that:
```
    let mut file = OpenOptions::new().write(true).truncate(true).create(true).open(&output_file).expect("failed to open output file");
    file.seek(SeekFrom::Current(mem::size_of::<u16>() as i64)).expect("failed to seek in output file");
    {
        let mut stdwrite = StdWrite(&mut file);
        header.to_wire(&mut stdwrite).expect("failed to write header");
        req.to_wire(&mut stdwrite).expect("failed to write request");
    }
    file.seek(SeekFrom::Start(0)).expect("failed to seek to beginning of output file");
    ...
```

Reply:
Remember that if `W: Write`, then `&mut W: Write`, too. Therefore, if you don't want to move into the `StdWrite`, you can pass in a mutable reference into it, resulting in a `StdWrite<&'a mut W>`.

Reply:
Ah yes, thank you!